File: ImpIntentApp2-main/ImpIntentApp2-main/app/src/main/java/com/inge/impintentapp2/MainActivity.java
package com.inge.impintentapp2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    EditText name;
    Button login;
    private String KEY_NAME = "NAMA";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        name = (EditText) findViewById(R.id.nama);
        login = (Button) findViewById(R.id.login);
        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    String nama = name.getText().toString();
                    if (nama != ""){
                        Intent i = new Intent(MainActivity.this, Welcome.class);
                        i.putExtra(KEY_NAME, nama);
                        startActivity(i);
                    } else {
                        Toast.makeText(getApplication(), "YOU NEED TO FILL YOUR NAME",Toast.LENGTH_SHORT);
                    }
                } catch (Exception e){
                    e.printStackTrace();
                    Toast.makeText(getApplication(), "ERROR, TRY AGAIN !",Toast.LENGTH_SHORT);
                }
            }
        });
    }
}

File: Manajemen_File/Manajemen_File/app/src/main/java/com/example/manajemen_file/MainActivity.kt
package com.example.manajemen_file

import android.Manifest
import android.content.Intent
import android.os.Bundle
import android.os.Environment
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import java.io.File
import java.io.FileOutputStream
import java.io.IOException
import java.lang.Exception

class MainActivity : AppCompatActivity() {
    var editText: EditText? = null
    private val STORAGE_PERMISSION_CODE = 23
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        editText = findViewById<View>(R.id.editText2) as EditText
    }

    fun next(view: View?) {
        val intent = Intent(this@MainActivity, activity_main2::class.java)
        startActivity(intent)
    }

    fun savePublic(view: View?) {
        //permission to access external storage
        ActivityCompat.requestPermissions(
            this,
            arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE),
            STORAGE_PERMISSION_CODE
        )
        val info = editText!!.text.toString()
        val folder =
            Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) //Folder Name
        val myFile = File(folder, "myData1.txt") //Filename
        writeData(myFile, info)
        editText!!.setText("")
    }

    fun savePrivate(view: View?) {
        val info = editText!!.text.toString()
        val folder = getExternalFilesDir("Tanti") //Folder Name
        val myFile = File(folder, "myData2.txt") //Filename
        writeData(myFile, info)
        editText!!.setText("")
    }

    private fun writeData(myFile: File, data: String) {
        var fileOutputStream: FileOutputStream? = null
        try {
            println("TES")
            fileOutputStream = FileOutputStream(myFile)
            fileOutputStream.write(data.toByteArray())
            Toast.makeText(this, "Done" + myFile.absolutePath, Toast.LENGTH_SHORT).show()
        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            if (fileOutputStream != null) {
                try {
                    fileOutputStream.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }
        }
    }
}